src/runtime/virtcontainers/sandbox.go

```diff
@@ -2100,9 +2100,8 @@ func (s *Sandbox) updateResources(ctx context.Context) error {
 	s.Logger().Debugf("Request to hypervisor to update oldCPUs/newCPUs: %d/%d", oldCPUs, newCPUs)
 	// If the CPUs were increased, ask agent to online them
 	if oldCPUs < newCPUs {
-		vcpusAdded := newCPUs - oldCPUs
-		s.Logger().Debugf("Request to onlineCPUMem with %d CPUs", vcpusAdded)
-		if err := s.agent.onlineCPUMem(ctx, vcpusAdded, true); err != nil {
+		s.Logger().Debugf("Request to onlineCPUMem with %d CPUs", newCPUs)
+		if err := s.agent.onlineCPUMem(ctx, newCPUs, true); err != nil {
```

Comment:
This modification is for compatibility with dragonball (which will do the online operation for us). 

The original meaning: num of vcpus we should do online operation this time.
Meaning after this modification: the total num of vcpus being onlined we should guaranteed after this operation.

If we do not modify this rpc, and we do not want to judge which vmm we are currently using, since dragonball may online part of the vcpus for us, the agent may not be able to online a sufficient number of vcpus when do the online operation. This may cause an error to be reported almost every time the agent does an online operation when dragonball is used as the hypervisor.